crates/syntax/src/ast/node_ext.rs

```diff
@@ -56,6 +56,10 @@ impl ast::BlockExpr {
     pub fn is_empty(&self) -> bool {
         self.statements().next().is_none() && self.tail_expr().is_none()
     }
+
+    pub fn as_lone_tail(&self) -> Option<ast::Expr> {
```

Comment:
I don't think that's the best place for this functionality -- one of the constraints of AST is that it should provide only "general" functionality. Specific helpers should generally be defined in the IDE layer. 

See, eg, `unwrap_trivial_block`. 

I suggest moving this elsewhere (and maybe amending the doc comment at the top explaining what should, and what should not go here).